File: Ai/ai_cpp/ai.cpp
#include "ai_client.h"
#include "card.h"

class AI : public AiClient
{
public:
    //选择初始卡组
    void chooseCards()
    {
        // artifacts和creatures可以修改
        my_artifacts = {"HolyLight"};
        my_creatures = {"Archer", "Swordsman", "Priest"};
        init();
    }

    void play()
    {
        if (round < 20)
            endRound();
        else
            exit(0);
    }
};

int main()
{
    AI player_ai;
    player_ai.chooseCards();
    while (true)
    {
        player_ai.updateGameInfo();
        player_ai.play();
    }
    return 0;
}

File: Ai/ai_py/ai.py
'''
AI
'''

from ai_client import AiClient


class AI(AiClient):
    '''玩家编写的ai
    '''

    def __init__(self):
        super().__init__()
        # 以下是玩家自己的初始构造

    def choose_cards(self):
        '''(根据初始阵营)选择初始卡组
        '''
        # artifacts和creatures可以修改
        self.artifacts = ["HolyLight"]
        self.creatures = ["Archer", "Swordsman", "VolcanoDragon"]
        self.init()

    def play(self):
        '''玩家需要编写的ai操作函数
        '''
        if self.round < 20:
            self.end_round()
        else:
            exit(0)


def main():
    '''启动AI
    '''
    player_ai = AI()
    player_ai.choose_cards()
    while True:
        player_ai.update_game_info()
        player_ai.play()


if __name__ == '__main__':
    main()
